### PR TITLE
Auto clean sensitive clipboard items

### LINKED ISSUE #2610: Auto-delete sensitive clip data in clipboard history
### Feature idea

Hi all,
New with Florisboard. Excellent project and keyboard. 
I noticed when copying from password manager and you have clipboard history enabled (plus the sync function) whatever you copied from PM wont be deleted from the history. 
I set the clipboard history to autodelete on 10 minutes, so no big deal but probably you may consider to build some interactions or add a section where you can restrict apps to save history. 
I use ditto on windows and have restricted keepassxc to track history and works well.

Thanks for yiur hard work on this.

### PR DESCRIPTION
**Pull request by lm41**
Closes #2610

### CODE REVIEW

File: app/src/main/kotlin/dev/patrickgold/florisboard/app/AppPrefs.kt [hunk 1]
```diff
@@ -118,6 +118,14 @@ class AppPrefs : PreferenceModel("florisboard-app-prefs") {
             key = "clipboard__clean_up_after",
             default = 20,
         )
+        val autoCleanSensitive = boolean(
+            key = "clipboard__auto_clean_sensitive",
+            default = false
+        )
+        val autoCleanSensitiveAfter = int(
+            key = "clipboard__auto_clen_sensitive_after",
```
[the remark this answers is not in the document]
Comment (lm41) [reply app/src/main/kotlin/dev/patrickgold/florisboard/app/AppPrefs.kt]: Thank you for noticing the typo :D

File: app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/clipboard/ClipboardScreen.kt [hunk 2]
```diff
@@ -71,6 +72,22 @@ fun ClipboardScreen() = FlorisScreen {
...
+                enabledIf = { prefs.clipboard.historyEnabled isEqualTo true },
+                visibleIf = { AndroidVersion.ATLEAST_API33_T },
+            )
+            DialogSliderPreference(
+                prefs.clipboard.autoCleanSensitiveAfter,
+                title = "Auto clean sensitive items after",
+                valueLabel = { pluralsRes(R.plurals.unit__seconds__written, it, "v" to it) },
+                min = 0,
+                max = 300,
+                stepIncrement = 5,
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/clipboard/ClipboardScreen.kt:87]:
```suggestion
                stepIncrement = 10,
```

I feel like with 10 it is easier to set to the desired value and it is still fine-grained enough for custom values

**Review by patrickgold — CHANGES_REQUESTED** [review]
Thanks a lot for your changes!

I have reviewed the code changes and have added comments where I'd change/nitpick stuff.

I have one not-yet-tested concern though: What happens when e.g. cleanUpOldAfter=1 and autoCleanSensitiveAfter=60 and both trigger to delete the same clip item concurrently, will it then crash? I currently don't have a device where I can reproduce this exact scenario, maybe you could test that?

File: app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/clipboard/ClipboardScreen.kt [hunk 3]
```diff
@@ -71,6 +72,22 @@ fun ClipboardScreen() = FlorisScreen {
                 stepIncrement = 5,
                 enabledIf = { prefs.clipboard.historyEnabled isEqualTo true && prefs.clipboard.cleanUpOld isEqualTo true },
             )
+            SwitchPreference(
+                prefs.clipboard.autoCleanSensitive,
+                title = "Auto clean sensitive items",
```
[the remark this answers is not in the document]
Comment (lm41) [reply app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/clipboard/ClipboardScreen.kt:77]: Oopsi :grin:

**Review by patrickgold — APPROVED** [review]
thanks for the fixes, lgtm now

**Pull request merged**